### PR TITLE
dbeaver/pro#2902 EAP limitations

### PR DESCRIPTION
**Pull request by ShadelessFox**

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ui.editors.connection/src/org/jkiss/dbeaver/ui/actions/datasource/DataSourceHandler.java [hunk 1]
```diff
@@ -89,6 +94,24 @@ public static void connectToDataSource(
                 return;
             }
 
+            if (DBWorkbench.getPlatform().getApplication().isEarlyAccessProgramActive()) {
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ui.editors.connection/src/org/jkiss/dbeaver/ui/actions/datasource/DataSourceHandler.java]: The idea was to limit total amount of connections, not just active ones.
We must check it in new connection create handler.
We also must check that only one project is loaded and restrict creation of more than one project.

Also we cannot use getAllDataSources function (it triggers all projects loading).

File: plugins/org.jkiss.dbeaver.ui.statistics/src/org/jkiss/dbeaver/ui/statistics/UIStatisticsActivator.java [hunk 2]
```diff
@@ -53,21 +51,6 @@ public static void setTrackingEnabled(boolean enabled) {
             return;
         }
         setPreferenceValue(PREF_FEATURE_TRACKING_ENABLED, enabled);
-        if (tracker != null) {
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ui.statistics/src/org/jkiss/dbeaver/ui/statistics/UIStatisticsActivator.java:56]: Why delete this code?

Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ui.statistics/src/org/jkiss/dbeaver/ui/statistics/UIStatisticsActivator.java:56]: isTrackingEnabled must return true if we are in EAP mode

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.ui.statistics/src/org/jkiss/dbeaver/ui/statistics/UIStatisticsActivator.java (hunk 2 above)
Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.ui.statistics/src/org/jkiss/dbeaver/ui/statistics/UIStatisticsActivator.java:56]: Unused.

Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.ui.statistics/src/org/jkiss/dbeaver/ui/statistics/UIStatisticsActivator.java:56]: The application instance is not available at this point. During the application startup, the preference is set to true, effectively emulating the desired behavior.

File: plugins/org.jkiss.dbeaver.ui.editors.connection/src/org/jkiss/dbeaver/ui/actions/datasource/DataSourceHandler.java (hunk 1 above)
Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.ui.editors.connection/src/org/jkiss/dbeaver/ui/actions/datasource/DataSourceHandler.java]: We haven't finished discussing the desired approach yet. This is the first iteration, and I think it's ok to keep it like this for now

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/app/DBPApplication.java [hunk 3]
```diff
@@ -136,4 +136,24 @@ default boolean isCommunity() {
         return false;
     }
 
+    /**
+     * Whether this product can participate in the Early Access Program but no license is present
+     */
+    default boolean isEarlyAccessProgramCandidate() {
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/app/DBPApplication.java]: WE don't use it in CE

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/app/DBPApplication.java [hunk 4]
```diff
@@ -136,4 +136,24 @@ default boolean isCommunity() {
...
+        return false;
+    }
+
+    /**
+     * Whether this product is part of the Early Access Program
+     */
+    default boolean isEarlyAccessProgramActive() {
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/app/DBPApplication.java]: Maybe in CE it should be isStatisticsCollectionRequired? Or move all stuff related to it into PRO

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.ui.statistics/src/org/jkiss/dbeaver/ui/statistics/PrefPageUsageStatistics.java [hunk 5]
```diff
@@ -76,6 +77,12 @@ protected Control createPreferenceContent(@NotNull Composite parent) {
             NLS.bind(StatisticCollectionMessages.statistic_collection_pref_content_opensource_link, LINK_GIHUB_REPO),
             SelectionListener.widgetSelectedAdapter(selectionEvent -> ShellUtils.launchProgram(LINK_GIHUB_REPO)));
 
+        if (DBWorkbench.getPlatform().getApplication().isStatisticsCollectionRequired()) {
+            checkSendUsageStatistics.setEnabled(false);
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ui.statistics/src/org/jkiss/dbeaver/ui/statistics/PrefPageUsageStatistics.java:81]: true?

Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.ui.statistics/src/org/jkiss/dbeaver/ui/statistics/PrefPageUsageStatistics.java:81]: It's selected down below in the `performDefaults` method

**Pull request merged**